///////////////////////////////////////////////////////////////////////////
// Copyright (C) Flowbox, Inc / All Rights Reserved
// Unauthorized copying of this file, via any medium is strictly prohibited
// Proprietary and confidential
// Flowbox Team <contact@flowbox.io>, 2014
///////////////////////////////////////////////////////////////////////////

package generated.proto.pluginManager;

import "plugin.proto";



message Plugin {
    message Add {
        message Request { 
            required generated.proto.plugin.Plugin plugin = 1;
        }
        message Update { 
            required Request request = 1;
            required int32 id = 2;
        }
    }
    message Remove {
        message Request { 
            required int32 id = 1;
        }
        message Update { 
            required Request request = 1;
        }
    }
    message Start {
        message Request { 
            required int32 id = 1;
        }
        message Update {
            required Request request = 1;
        }
    }
    message Stop {
        message Request { 
            required int32 id = 1;
        }
        message Update { 
            required Request request = 1;
        }
    }
    message Restart {
        message Request { 
            required int32 id = 1;
        }
        message Update { 
            required Request request = 1;
        }
    }
    message Lookup {
        message Request { 
            required int32 id = 1;
        }
        message Status { 
            required Request request = 1;
            required generated.proto.plugin.PluginInfo pluginInfo = 2;
        }
    }
    message List {
        message Request { }
        message Status { 
            required Request request = 1;
            repeated generated.proto.plugin.PluginInfo list = 2;
        }
    }
}

message PluginManager {
    message Ping {
        message Request { }
        message Status  { 
            required Request request = 1;
        }
    }
}